src/Adapter/ImageThumbnailsRegenerator.php

```diff
@@ -0,0 +1,310 @@
+                }
+            }
+        }
+
+        // Delete product images using new filesystem.
+        if ($isProduct) {
+            $productsImages = $this->productImageRepository->getAllImages();
+            foreach ($productsImages as $image) {
+                if (file_exists($dir . $image->getImgFolder())) {
+                    $toDel = scandir($dir . $image->getImgFolder(), SCANDIR_SORT_NONE);
+                    foreach ($toDel as $d) {
+                        foreach ($types as $imageType) {
+                            if (preg_match('/^[0-9]+\-' . $imageType->getName() . '(|2x)\.(jpg|png|webp|avif)$/', $d)
+                                || (count($types) > 1 && preg_match('/^[0-9]+\-[_a-zA-Z0-9-]*\.(jpg|png|webp|avif)$/', $d))) {
+                                if (file_exists($dir . $image->getImgFolder() . $d)) {
+                                    unlink($dir . $image->getImgFolder() . $d);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Regenerate images.
+     *
+     * @param string $dir
+     * @param array $type
+     * @param bool $productsImages
+     *
+     * @return bool|array
+     */
+    public function regenerateNewImages(string $dir, array $type, bool $productsImages = false): bool|array
+    {
+        if (!is_dir($dir)) {
+            return false;
+        }
+
+        $errors = [];
+
+        /*
+         * Let's resolve which formats we will use for image generation.
+         *
+         * In case of .jpg images, the actual format inside is decided by ImageManager.
+         */
+        $configuredImageFormats = $this->imageFormatConfiguration->getGenerationFormats();
+
+        if (!$productsImages) {
+            $formated_medium = LegacyImageType::getFormattedName('medium');
+            foreach (scandir($dir, SCANDIR_SORT_NONE) as $image) {
+                if (preg_match('/^[0-9]*\.jpg$/', $image)) {
+                    foreach ($type as $k => $imageType) {
+                        // Customizable writing dir
+                        $newDir = $dir;
+                        if (!file_exists($newDir)) {
+                            continue;
+                        }
+
+                        if (($dir == _PS_CAT_IMG_DIR_) && ($imageType->getName() == $formated_medium) && is_file(_PS_CAT_IMG_DIR_ . str_replace('.', '_thumb.', $image))) {
+                            $image = str_replace('.', '_thumb.', $image);
+                        }
+
+                        foreach ($configuredImageFormats as $imageFormat) {
+                            // For JPG images, we let Imagemanager decide what to do and choose between JPG/PNG.
+                            // For webp and avif extensions, we want it to follow our command and ignore the original format.
+                            $forceFormat = ($imageFormat !== 'jpg');
```

Comment:
Please check the current code, there have been some changes in https://github.com/PrestaShop/PrestaShop/pull/34172. :-)